src/Sentry/Scope.cs

```diff
@@ -738,6 +780,24 @@ public void AddAttachment(string filePath, AttachmentType type = AttachmentType.
                 Path.GetFileName(filePath),
                 contentType));
 
-    internal void ResetTransaction(ITransactionTracer? expectedCurrentTransaction) =>
-        Interlocked.CompareExchange(ref _transaction, null, expectedCurrentTransaction);
+    /// <summary>
+    /// We need this lock to prevent a potential race condition in <see cref="ResetTransaction"/>.
+    /// </summary>
+    private readonly ReaderWriterLockSlim _transactionLock = new();
+
+    internal void ResetTransaction(ITransactionTracer? expectedCurrentTransaction)
+    {
+        _transactionLock.EnterWriteLock();
+        try
+        {
+            if (_transaction.Value == expectedCurrentTransaction)
```

Comment:
Instead of `==` (operator) can we go with `object.ReferenceEquals(_transaction.Value, expectedCurrentTransaction);`? (iirc that's the API)

Reply:
Done... out of interest, why is this preferable? Is this in case someone messes with the equality operator on their `ITransactionTracer` implementation)?

Reply:
yeah one is an operator that can be overridden, and the other simply has no doubt of what it is or how it works.